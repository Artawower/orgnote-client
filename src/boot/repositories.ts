import { boot } from 'quasar/wrappers';
import { Database, NoteRepository, FileRepository } from '../repositories';

const db = new Database(NoteRepository, FileRepository);

const repositories = {
  notes: new NoteRepository(db),
  files: new FileRepository(db),
} as const;

export default boot(({ app }) => {
  app.config.globalProperties.$db = db;
  app.config.globalProperties.$repositories = repositories;
});

export { db, repositories };
