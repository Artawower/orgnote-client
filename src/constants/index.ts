export * from './initial-note-template';
export * from './image-file-extensions';
