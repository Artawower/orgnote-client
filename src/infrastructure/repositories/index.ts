import type { OrgNoteApi } from 'orgnote-api';
import { createDatabase } from './create-database';
import { NOTE_REPOSITORY_NAME, NOTE_MIGRATIONS } from './note-info-repository';
import { FILE_REPOSITORY_NAME, FILE_MIGRATIONS } from './file-repository';
import { createFileRepository } from './file-repository';
import { createNoteInfoRepository } from './note-info-repository';

export async function initRepositories(): Promise<OrgNoteApi['infrastructure']> {
  const { db } = createDatabase([
    { storeName: NOTE_REPOSITORY_NAME, migrations: NOTE_MIGRATIONS },
    { storeName: FILE_REPOSITORY_NAME, migrations: FILE_MIGRATIONS },
  ]);
  return {
    fileInfoRepository: createFileRepository(db),
    noteInfoRepository: createNoteInfoRepository(db),
  };
}
