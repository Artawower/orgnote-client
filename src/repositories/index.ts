export * from './note-repository';
export * from './repository';
export * from './database';
export * from './file-repository';
export * from './orgnote-database';
