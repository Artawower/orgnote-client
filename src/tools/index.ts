export * from './extract-file';
export * from './quill-formatter';
export * from './tinykeys-wrapper';
export * from './quill-blots';
export * from './generate-file-name';
export * from './remove-file-extension';
export * from './debounce';
export * from './is-org-file';
export * from './read-file';
export * from './is-url';
export * from './get-file-name';
export * from './sleep';
