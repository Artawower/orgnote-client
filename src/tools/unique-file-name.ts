export const getUniqueFileName = (
  existingFileNames: string[],
  fileExt = '.org',
  filePrefix = 'untitled-note'
): string => {
  let initialName = `${filePrefix}${fileExt}`;

  let inc = 0;
  while (existingFileNames.includes(`${initialName}`)) {
    inc++;
    initialName = `${filePrefix}-${inc}${fileExt}`;
  }

  return initialName;
};
