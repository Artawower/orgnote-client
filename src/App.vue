<template>
  <router-view />
</template>

<script lang="ts">
export default {
  name: 'App',
};
</script>

<script lang="ts" setup>
import { useRouter } from 'vue-router';
import { jsBabel } from './plugins/js-babel';
import { RouteNames } from './router/routes';
import {
  useAuthStore,
  useOrgBabelStore,
  useLoggerStore,
  useSyncStore,
  useOrgNoteApiStore,
} from './stores';
import { decodeAuthState, extractAuthQueryInfo } from './tools';

const syncStore = useSyncStore();

useLoggerStore().init();

syncStore.markToSync();

const orgBabelStore = useOrgBabelStore();
orgBabelStore.register(jsBabel);

if (window.navigator.standalone) {
  document.body.classList.add('standalone');
}

const authStore = useAuthStore();

const router = useRouter();

// TODO: master move to external method.
async function handleCordovaAuth(url: string) {
  const urlParams = url.split('?')?.[1];
  if (!urlParams) {
    return;
  }
  const searchParams = new URLSearchParams(urlParams);

  const personalInfo = extractAuthQueryInfo(Object.fromEntries(searchParams));
  console.debug('✎: [AUTH] personalInfo: ', personalInfo);
  const state = decodeAuthState(searchParams.get('state'));
  console.debug('✎: [AUTH] state: ', state);

  await authStore.authUser(personalInfo, searchParams.get('token'));

  if (state.redirectUrl) {
    window.location.assign(`/#${state.redirectUrl}`);
    return;
  }

  router.push({ name: RouteNames.Home });
}
(window as unknown as { handleOpenURL: (arg0: string) => void }).handleOpenURL =
  handleCordovaAuth.bind(this);

const { orgNoteApi } = useOrgNoteApiStore();
window.orgnote = orgNoteApi;
</script>
