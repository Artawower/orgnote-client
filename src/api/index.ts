import { Note } from 'src/models';

export interface OrgNoteApi {
  navigation: {
    openNote: (id: string) => void;
    editNote: (id: string) => void;
  };
  currentNote: {
    getCurrentNote: () => Note;
  };
  configuration: () => OrgNoteConfig;
}

// NOTE: default configs
export interface OrgNoteConfig {
  editor: {
    showSpecialChars: boolean;
  };
}
