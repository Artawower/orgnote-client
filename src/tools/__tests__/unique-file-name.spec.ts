import { FileTree } from 'src/repositories';
import { getUniqueFileName } from '../unique-file-name';

describe('unique-file-name', () => {
  it('should return unique name', () => {
    const fileNode: FileTree = {
      'Untitled-note.org': {
        name: 'Untitled-note.org',
        filePath: ['d1'],
        type: 'file',
      },
    };

    const fileName = getUniqueFileName(fileNode);

    expect(fileName).toMatchSnapshot();
  });

  it('Should return unique name with multiple default names', () => {
    const fileNode: FileTree = {
      'Untitled-note.org': {
        name: 'Untitled-note.org',
        filePath: ['d1'],
        type: 'file',
      },
      'Untitled-note-1.org': {
        name: 'Untitled-note-1.org',
        filePath: ['d1'],
        type: 'file',
      },
      'Untitled-note-2.org': {
        name: 'Untitled-note-2.org',
        filePath: ['d1'],
        type: 'file',
      },
    };
    const fileName = getUniqueFileName(fileNode);

    expect(fileName).toMatchSnapshot();
  });
});
