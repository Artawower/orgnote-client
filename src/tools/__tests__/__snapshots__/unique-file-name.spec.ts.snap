// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`unique-file-name should return unique name 1`] = `"Untitled-note-1"`;

exports[`unique-file-name Should return unique name with multiple default names 1`] = `"Untitled-note-3"`;
